lib/krane/resource_deployer.rb

```diff
@@ -95,11 +95,10 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       applyables, individuals = resources.partition { |r| r.deploy_method == :apply }
       # Prunable resources should also applied so that they can  be pruned
       pruneable_types = @prune_whitelist.map { |t| t.split("/").last }
-      applyables += individuals.select { |r| pruneable_types.include?(r.type) }
+      applyables += individuals.select { |r| pruneable_types.include?(r.type) && !r.deploy_method_override }
```

Comment:
We need to fight against default behaviour, which makes me loath to ever merge this into master. Previously, we would run `apply` here to enforce pruning, but since the goal right now is to _avoid_ pruning, we need to add this workaround. 

The good news is that this _only_ affects resources that are purposefully overridden, but I think it's enough of a landmine to give me some pause

Reply:
Would It be helpful to ever see a log warning when a pruneable resource is being overridden?   I think if it's purposely overridden, then they know they're fighting the default. but wondering if it would give less pause and help to debug if that knowledge assumption is wrong.